### DISCUSSION TITLE
Doubt regarding Input file type

### DISCUSSION THREAD

**Discussion by Anurag19101996** [uw-ssec/MAWpy#38] [General]
The current code is only operable for a dataframe type input file (tabular in nature). But we’d like the file to take in other data types such as json, other tabular ones (like SQL, xlsx) etc. Will the functionalities to support other data types be added by Escience?

@qzchen-uw @gracejia513 @anujsinha3 @carlosgjs

**Comment by anujsinha3** [comment]
As per the initial discussions, we expect the code to have a standardized input type, as well as column nomenclature for improved performance. That being said, the code can handle .csv file format which is quite interoperable and pythonic format. Also, formats like xlsx can be easily be converted to csv before being passed as input to the code.

**Reply by gracejia513** [reply]
@anujsinha3 It sounds like the user has to do their own conversion to translate their data in to csv and csv only before using our workflow. And the current workflow1 also strictly only allow csv file after testing the functions. @qzchen-uw do you prefer the workflow to accept a variety of input types or let the user do the conversion on their own?

**Reply by qzchen-uw** [reply]
@gracejia513 @anujsinha3 I think we shall allow accepting a variety of input files including csv files. But workflow1 is already done?

**Reply by anujsinha3** [reply]
Sure, Let's discuss this in our next meeting.

**Comment by gracejia513** [comment]
@anujsinha3 another issue is that workflow1/intermental_clustering.py does not work with csv files containing only the four required columns, USER_ID, ORIG_LAT, ORIG_LONG, and UNIX_START_T. The algorithm returns an error message (see below):
![image](https://github.com/user-attachments/assets/2ce37239-c8c0-43e5-8f14-f22658f3b01e)

This suggests that the input file structure is not correctly specified. And I have another screenshot of the input data I used (see below):
![image](https://github.com/user-attachments/assets/1468f279-4071-4c4c-8915-2ec3ad5998dd)

Can you please look into this?

**Reply by anujsinha3** [reply]
USER_ID, ORIG_LAT, ORIG_LONG, and UNIX_START_T are variables that correspond to lowercase strings. It's a programming convention to use CAPITAL CASE variables for string constants. Refer to the file [here](https://github.com/uw-ssec/MAWpy/blob/main/src/mawpy/constants.py):

USER_ID = 'user_ID'
ORIG_LAT = 'orig_lat'
ORIG_LONG = 'orig_long'
UNIX_START_T = 'unix_start_t'

In short, the correct way is to have column names corresponding to the values of these variables (small snake cased strings)

**Reply by gracejia513** [reply]
I fixed the column names to lower case, and there is still an error: 
<img width="817" alt="image" src="https://github.com/user-attachments/assets/bc4a3cab-99e9-4a16-9bdf-ae9aaf38bfc5">
i'm also attaching my input file, could @anujsinha3 and @Anurag19101996 check if the error persists on your end? 
[input_file_4_columns.csv](https://github.com/user-attachments/files/16228704/input_file_4_columns.csv)
